openwisp_controller/config/admin.py

```diff
@@ -20,9 +20,14 @@ class ConfigForm(AlwaysHasChangedMixin, AbstractConfigForm):
     class Meta(AbstractConfigForm.Meta):
         model = Config
 
+    device_model = Device
+
     def clean_templates(self):
         org = Organization.objects.get(pk=self.data['organization'])
-        self.cleaned_data['organization'] = org
+        device = self.device_model(name=self.data['name'],
+                                   organization=org,
+                                   mac_address=self.data['mac_address'])
+        self.cleaned_data['device'] = device
```

Comment:
I'm not really sure this method makes sense anymore if we remove the organization field from `Config`, try removing it altogether. I don't even understand what is being done here and why.